Cython/Utility/Profile.c

```diff
@@ -103,7 +113,20 @@
   CYTHON_UNUSED static int __Pyx__TraceResumeGen(PyMonitoringState *state_array, __pyx_monitoring_version_type *version, PyObject *code_obj, int offset); /*proto*/
   CYTHON_UNUSED static void __Pyx__TraceException(PyMonitoringState *monitoring_state, PyObject *code_obj, int offset, int reraised); /*proto*/
 
-  #define __Pyx_PyMonitoring_ExitScope()  (PyMonitoring_ExitScope(), (void) __pyx_exception_already_reported)
+  #define __Pyx_PyMonitoring_ExitScope(nogil)              \
+    (void) __pyx_exception_already_reported;               \
+    if (nogil) {                                           \
```

Comment:
`__Pyx_PyMonitoring_ExitScope()` looks like a single statement, so I think we should also implement it as such.
```suggestion
  #define __Pyx_PyMonitoring_ExitScope(nogil)              \
    if (nogil) {                                           \
        (void) __pyx_exception_already_reported;           \
```